File: challenge2/src/main/java/com/example/demo/Services/Interfaces/IItemService.java
package com.example.demo.Services.Interfaces;

import com.example.demo.Dtos.ItemDto;

import java.util.List;

public interface IItemService {
  List<ItemDto> getItems();

  ItemDto getItemById(long id);

  ItemDto createItem(ItemDto dto);

  ItemDto updateItem(long id, ItemDto dto);

  void deleteItem(long id);
}

File: py-example/app/service/iitem_service.py
from typing import List
from ..dto.item_dto import ItemCreateDTO, ItemUpdateDTO, ItemDTO

class IItemService:
    def create_item(self, item_create_dto: ItemCreateDTO) -> ItemDTO:
        raise NotImplementedError

    def get_item(self, item_id: int) -> ItemDTO:
        raise NotImplementedError

    def get_items(self) -> List[ItemDTO]:
        raise NotImplementedError

    def update_item(self, item_id: int, item_update_dto: ItemUpdateDTO) -> ItemDTO:
        raise NotImplementedError

    def delete_item(self, item_id: int):
        raise NotImplementedError